packages/odyssey-react-mui/src/Radio.tsx

```diff
@@ -95,8 +96,23 @@ const Radio = ({
   );
 
   const onChange = useCallback<NonNullable<MuiRadioProps["onChange"]>>(
-    (event, checked) => onChangeProp?.(event, checked),
-    [onChangeProp],
+    (event, checked) => {
+      if (isReadOnly) {
+        event.preventDefault();
+        return;
+      }
+      onChangeProp?.(event, checked);
+    },
+    [onChangeProp, isReadOnly],
+  );
+
+  const handleMouseDown = useCallback<React.MouseEventHandler<HTMLSpanElement>>(
```

Comment:
This should probably be a click handler, rather than mousedown. Can still click with the keyboard